shopping-order-service/src/main/java/shopping/order/service/OrderServiceImpl.java

```diff
@@ -29,6 +29,20 @@ public OrderResponse createOrder(OrderRequest orderRequest) {
 		return toOrderResponse(order);
 	}
 
+	@Override
+	public OrderResponse getOrderById(String productId) {
+		return toOrderResponse(repository.findById(productId).orElseThrow(() -> new RuntimeException("Not fount order "+productId)));
+	}
+
+	@Override
+	public OrderResponse putOrder(String id, OrderRequest orderRequest) {
```

Comment:
new data of the requested order is not persisted to db. You have to convert the OrderRequest to Order entity first and then call the repository.save method to persist that new order.